### PR TITLE
Forms Automated Test - LPS-134377 | master

### PR DESCRIPTION
**Pull request by Diogomes**

### CODE REVIEW

**Comment by natocesarrego** [comment]
@Diogomes, the uploaded `index.html` on the Jira ticket doesn't provide the test execution duration neither if it passed or not, can you please check it and fix?

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase [hunk 1]
```diff
@@ -97,14 +97,61 @@ definition {
...
+			category = "Content &amp; Data",
+			portlet = "Forms");
+
+		FormsAdminNavigator.gotoAddForm();
+
+		Form.editName(formName = "Form 1");
+
+		Form.gotoAddField(
+			fieldPositionNumber = "0",
+			fieldType = "Text");
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase:115]: @Diogomes, please add a line break after this line

**Comment by Diogomes** [comment]
> @Diogomes , o upload `index.html`no tíquete do Jira não fornece a duração da execução do teste nem se passou ou não, você pode verificar e corrigir?

Yes I Can

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase [hunk 2]
```diff
@@ -97,14 +97,61 @@ definition {
...
+			positionNumber = "1",
+			value = "Text");
+
+		FormRules.viewConfigureRules(
+			positionNumber = "2",
+			value = "Text");
 
+		FormRules.viewConfigureRules(
+			positionNumber = "2",
+			value = "Is empty");
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase:154]: @Diogomes, can you please use the existing macros instead, like `FormRules.viewConditionFieldSummary`?

Please take a look on the macros available on `FormRules.macro`. :wink:

**Review by natocesarrego — CHANGES_REQUESTED** [review]

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase [hunk 3]
```diff
@@ -97,14 +97,61 @@ definition {
...
+
+		FormsAdminNavigator.gotoBack();
+
+		FormsAdminNavigator.gotoEditFormByName(formName = "Form 1");
+
+		FormsAdminNavigator.gotoRules();
+
+		Click(locator1 = "FormRules#RULE_BUILDER_ADD_RULE_BUTTON");
+
+		FormRules.viewConfigureRules(
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase:140]: @Diogomes, can we create two separated macros instead? Ideally, one will check the condition and the other one will check the action.

**Comment by Diogomes** [comment]
@natocesarrego  All done . 😄

File: portal-web/test/functional/com/liferay/portalweb/macros/FormRules.macro [hunk 4]
```diff
@@ -296,6 +296,20 @@ definition {
 			locator1 = "FormRules#RULE_BUILDER_BROKEN");
 	}
 
+	macro viewBuilderRuleAction {
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormRules.macro:299]: @Diogomes, please rename to `viewRuleBuilderAction` to keep consistency with the name used in the UI and in the path `RULE_BUILDER_FIELD_GENERIC`

File: portal-web/test/functional/com/liferay/portalweb/macros/FormRules.macro [hunk 5]
```diff
@@ -296,6 +296,20 @@ definition {
...
+		AssertElementPresent(
+			key_positionNumber = "${positionNumber}",
+			key_value = "${value}",
+			locator1 = "FormRules#RULE_BUILDER_FIELD_GENERIC");
+	}
+
+	macro viewBuilderRuleCondition {
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/macros/FormRules.macro:306]: @Diogomes, please rename to `viewRuleBuilderCondition` to keep consistency with the name used in the UI and in the path `RULE_BUILDER_FIELD_GENERIC`

File: portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase [hunk 6]
```diff
@@ -97,14 +97,62 @@ definition {
...
+			positionNumber = "1",
+			value = "Text");
+
+		FormRules.viewBuilderRuleAction(
+			positionNumber = "2",
+			value = "Text");
 
+		FormRules.viewBuilderRuleCondition(
+			positionNumber = "2",
+			value = "Is empty");
```
Comment (natocesarrego) [inline portal-web/test/functional/com/liferay/portalweb/tests/enduser/webformsanddatalists/forms/FormRules.testcase:155]: @Diogomes, let's perform the assertion of the whole condition then we perform the assertion of the whole action, like below:
```
FormRules.viewBuilderRuleCondition(
    positionNumber = "1",
    value = "Text");

FormRules.viewBuilderRuleCondition(
    positionNumber = "2",
    value = "Is empty");

FormRules.viewBuilderRuleAction(
    positionNumber = "1",
    value = "Require");

FormRules.viewBuilderRuleAction(
    positionNumber = "2",
    value = "Text");
```

**Review by natocesarrego — CHANGES_REQUESTED** [review]

**Comment by natocesarrego** [comment]
ci:forward

**Pull request closed without merging**